infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -179,6 +186,35 @@ public ServerContext(Map<String, Object> serverConfig) {
         nodeLocator = NodeLocator
                 .parseString(serverConfig.get("--address") + ":" + serverConfig.get("<port>"));
         localEndpoint = nodeLocator.toEndpointUrl();
+        streamNames = loadStreamIdToNameMap((String) serverConfig.get("--stream-names-file"));
+    }
+
+    public static Map<UUID, String> loadStreamIdToNameMap(String streamNameFile) {
+        Map<UUID, String> streamNames = new HashMap<>();
+        if (streamNameFile == null) {
```

Comment:
what if 'streamNameFile' is an empty string

Reply:
there will be an exception thrown and the map will be initialized to an empty map, which is the default case for all our internal ITs and unit tests.